infra/terraform/testground.tf

```diff
@@ -1,130 +1,13 @@
-  availability_zone       = "${var.aws_availability_zone}"
-  map_public_ip_on_launch = true
-}
-
-resource "aws_security_group" "default" {
-  name        = "testground"
-  description = "Testground network security rules"
-  vpc_id      = "${aws_vpc.default.id}"
-
-  ingress {
-    from_port   = 22
-    to_port     = 22
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+variable "default_tags" {
+  type = map
 
-  ingress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["172.16.0.0/16"]
-  }
-
-  ingress {
-    from_port   = 60000
-    to_port     = 61000
-    protocol    = "udp"
-    cidr_blocks = ["0.0.0.0/0"]
-    description = "mosh"
-  }
-
-  # outbound internet access
-  egress {
-    from_port = 0
-
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-}
-
-resource "aws_instance" "testground_manager" {
-  ami                    = "${lookup(var.aws_amis, var.aws_region)}"
-  availability_zone      = "${var.aws_availability_zone}"
-  instance_type          = "${var.aws_instance_type_manager}"
-  key_name               = "${var.key_name}"
-  vpc_security_group_ids = ["${aws_security_group.default.id}"]
-  iam_instance_profile   = "EC2-Full-API-Access-Dangerous"
-  subnet_id              = "${aws_subnet.default.id}"
-  private_ip             = "172.16.0.10"
-
-  tags = {
-    TG     = "${var.tag}"
-    Name   = "${var.tag}-manager"
-    TGRole = "manager"
+  default = {
+    Project = "testground"
   }
 }
 
-resource "aws_instance" "testground_redis" {
-  ami                    = "${lookup(var.aws_amis, var.aws_region)}"
-  availability_zone      = "${var.aws_availability_zone}"
-  instance_type          = "${var.aws_instance_type_redis}"
-  key_name               = "${var.key_name}"
-  vpc_security_group_ids = ["${aws_security_group.default.id}"]
-  iam_instance_profile   = "EC2-Full-API-Access-Dangerous"
-  subnet_id              = "${aws_subnet.default.id}"
-  private_ip             = "172.16.0.11"
-
-  tags = {
-    TG     = "${var.tag}"
-    Name   = "${var.tag}-redis"
-    TGRole = "redis"
-  }
-}
-
-module "asg" {
-  source = "terraform-aws-modules/autoscaling/aws"
-
-  name = "service"
-
-  # Launch configuration
-  lc_name = "${var.tag}"
-
-  image_id             = "${lookup(var.aws_amis, var.aws_region)}"
-  instance_type        = "${var.aws_instance_type_worker}"
-  key_name             = "${var.key_name}"
-  security_groups      = ["${aws_security_group.default.id}"]
-  iam_instance_profile = "EC2-Full-API-Access-Dangerous"
-
-  # Auto scaling group
-  asg_name            = "${var.tag}"
-  vpc_zone_identifier = ["${aws_subnet.default.id}"]
-  health_check_type   = "EC2"
-  min_size            = "${var.workers}"
-  max_size            = "${var.workers}"
-  desired_capacity    = "${var.workers}"
-
-  tags_as_map = {
-    TG     = "${var.tag}"
-    Name   = "${var.tag}-worker"
-    TGRole = "worker"
-  }
+data "aws_ami" "base" {
+  most_recent = true
```

Comment:
I'm getting this error when I try to run:

```
Error: Error launching source instance: UnsupportedOperation: AMI 'ami-0f37d6c94b5686e2a' with an instance-store root device is not supported for the instance type 'c5.large'.
	status code: 400, request id: e405ffbf-9c25-433a-b1d1-b738358f3b68
```

I suspect the most recent AMI uploaded by Amazon supports the wrong type of root device. There's probably the ability to add an additional filter.

Reply:
I added this to get it to work:

```
filter {
    name   = "root-device-type"
    values = ["ebs"]
  }
```

Reply:
I suspect this `data` call is not returning the correct AMI based on the input instance type. I haven't tested yet with a different instance type by the default - thanks for noticing this, I will make sure it works with ebs, instance store, and all other types.